### DISCUSSION TITLE
Synchronised instruments

### DISCUSSION THREAD

**Discussion by jpsecher** [microsoft/Qcodes#4787] [General]
I want to make it possible for users to connect several identical instruments together to control the collective set of channels as one instrument.  The specific instrument is QDAC-II, and it has special features for synchronising the clocks of several units.

What would you suggest as the best route for implementing this?  Should I create something that fits in the `Instrument` hierarchy that delegates commands to underlying `VisaInstrument`s?

**Comment by jenshnielsen** [comment]
@jpsecher Yes I think I would go that route. We don't have a close matching example in QCoDeS but perhaps the virtual driver for the AMI magents could serve some inspiration. See AMIModel4303D [here](https://github.com/QCoDeS/Qcodes/blob/master/qcodes/instrument_drivers/american_magnetics/AMI430_visa.py) In this case the purely virtual instrumnent is responible for delegating commans to 3 magnet psus to allow the user to set and get magnetic fields in various meaningful coordinate systems without having to manually perform the transformations
